qermit/zero_noise_extrapolation/zne.py

```diff
@@ -75,14 +75,42 @@ def circuit(circ: Circuit, noise_scaling: int, **kwargs) -> Circuit:
             for gate in reversed(circ.get_commands()):
                 if gate.op.type == OpType.Barrier:
                     folded_circ.add_barrier(gate.args)
+                elif gate.op.type == OpType.CircBox:
```

Comment:
is that actually adding the inverse of the box? As a Box operation is not expected here, i'd do one "if gate.op.type == OpType.Box" or box_types = {OpType.CircBox, ... ,OpType.Unitary3qBox},"if gate.op.type in box_types: raise Error("doesnt support box types")"

Reply:
Probably the latter, i.e. don't add Box operations (you're not expecting them I believe), but at least make sure if they're encountered an error is thrown with readable string, else another less readable error will be.